src/ray/util/process.cc

```diff
@@ -636,6 +636,87 @@ bool IsProcessAlive(pid_t pid) {
 #endif
 }
 
+#if defined(__linux__)
+static inline std::error_code KillProcLinux(pid_t pid) {
+  std::error_code error;
+  if (kill(pid, SIGKILL) != 0) {
+    error = std::error_code(errno, std::system_category());
+  }
+  return error;
+}
+#endif
+
+std::optional<std::error_code> KillProc(pid_t pid) {
+#if defined(__linux__)
+  return {KillProcLinux(pid)};
+#else
+  return std::nullopt;
+#endif
+}
+
+#if defined(__linux__)
+static inline std::vector<pid_t> GetAllProcsWithPpidLinux(pid_t parent_pid) {
+  std::vector<pid_t> child_pids;
+
+  // Iterate over all files in the /proc directory, looking for directories.
+  // See `man proc` for information on the directory structure.
+  // Directories with only digits in their name correspond to processes in the process
+  // table. We read in the status of each such process and parse the parent PID. If the
+  // process parent PID is equal to parent_pid, then we add it to the vector to be
+  // returned. Ideally, we use a library for this, but at the time of writing one is not
+  // available in Ray C++.
+
+  std::filesystem::directory_iterator dir(kProcDirectory);
```

Comment:
does this work in docker container too (and is it configurable?)

Reply:
Yes, works in docker container. The [man pages](https://manpages.ubuntu.com/manpages/xenial/man5/proc.5.html) suggest that there is a way you can change it, but I think it's really rare.

> The  proc  filesystem  is  a  pseudo-filesystem which provides an interface to kernel data
       structures.  It is commonly mounted at [/proc](file:///proc).  Most of it is  read-only,  but  some  files
       allow kernel variables to be changed.